### ISSUE TITLE
CRT runtime optional inclusion

### ISSUE THREAD

**Issue by fffonion** [pyinstaller/pyinstaller#8851]
**Is your feature request related to a problem? Please describe.**

It seems the ucrtbase.dll and couple of other crt runtime dlls are always included
to maintain compatibility at win7 (read from multiple issues like #6326 #4605).

However, the latest version that supports win7 is Python3.8 (correct me if I'm wrong),
that means if I'm bundling my program with Python >3.8 version, my application won't
run on win7 anyways, thus I don't even need those crt runtimes.

**Describe the solution you'd like**
Give user the option to exclude crt runtime libraries to reduce file size. Or just automatically skip
them if the python version is larger than 3.8.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by bwoodsend** [comment]
There are older versions of Windows 10 that don't have UCRT (although you can install it).

**Comment by rokm** [comment]
> There are older versions of Windows 10 that don't have UCRT (although you can install it).

Isn't that Windows 8/8.1?

According to Microsoft, "The UCRT is now a Windows component, and ships as part of Windows 10 and later."

But either way, if user explicitly targets only Win10 and 11, it makes sense to be able to avoid collection of unnecessary UCRT DLLs.

**Comment by bwoodsend** [comment]
>> There are older versions of Windows 10 that don't have UCRT (although you can install it).
> 
> Isn't that Windows 8/8.1?

Sounds probable. Windows 8.1 isn't tied to Python <= 3.8 though (although all mention of Windows 8.0 or 8.1 seems to have disappeared from python.org/downloads) so implicitly excluding UCRT based on Python version is a no-go.

**Comment by robUx4** [comment]
I looked into adding an option for this but the inclusion/exclusion lists for DLLs are global variables in [depend/dylib.py](https://github.com/pyinstaller/pyinstaller/blob/c5eb50cfb41eae75c9afc5535cd633e349606c1b/PyInstaller/depend/dylib.py#L53). Adding an option to trigger doesn't seem to possible (without considerable hacking).

Given Windows 10 will soon be dead/unsupported, it may be time to assume all Windows machines have UCRT. Users who still need Windows 7/8/8.1 support can still use an older version of PyInstaller. The Windows 7 support seems to have been [dropped with Python 3.9.0](https://www.python.org/downloads/release/python-390/). Windows 8.1 will be [dropped in Python 3.14](https://docs.python.org/3.14/using/windows.html) too:

> As specified in [PEP 11](https://peps.python.org/pep-0011/), a Python release only supports a Windows platform while Microsoft considers the platform under extended support. This means that Python 3.14 supports Windows 10 and newer. If you require Windows 7 support, please install Python 3.8. If you require Windows 8.1 support, please install Python 3.12.

See https://github.com/python/cpython/commit/da1e5526aee674bb33c17a498aa3781587b9850c which is in the 3.14 tags, not the 3.13 so the documentation is slightly bogus. 3.13 still supports Windows 8.1 (which has the api-win-*** files once [updated](https://support.microsoft.com/en-us/topic/update-for-universal-c-runtime-in-windows-c0514201-7fe6-95a3-b0a5-287930f3560c)).

It seems Windows 8.0 was dropped in 3.12: https://github.com/python/cpython/commit/a4916e6013e0870fd682b7967dce76667e8ba547

**Comment by bwoodsend** [comment]
> It seems Windows 8.0 was dropped in 3.12: https://github.com/python/cpython/commit/a4916e6013e0870fd682b7967dce76667e8ba547

8.0 was blocked in the in Python 3.9 like Windows 7 was. (That change says as much in its news entry.)

> Given Windows 10 will soon be dead/unsupported, it may be time to assume all Windows machines have UCRT.

Honestly, I don't really care what Microsoft's official support window is. [Usage statistics](https://gs.statcounter.com/windows-version-market-share/desktop/worldwide) still say Windows 10 is still more popular than 11 and that you only need a pool of ~40 users before you encounter a Windows 7.

**Comment by robUx4** [comment]
I agree, although the removal of UCRT DLL's could happen if we detect Python 3.14 which will no longer (officially) run on Windows older than Windows 10. It's a bit early for that but in 6 months Python 3.14 will be a reality.

I'd still favor an optional flag, but I don't think it's feasible without considerable refactoring.
